.github/workflows/main.yml

```diff
@@ -0,0 +1,20 @@
+name: Collector CI
+
+on:
+  push:
+    branches:
+      - master
+      - release-*
+  pull_request:
+    branches:
+      - master
```

Comment:
I've removed this limitation now, but we'll have to keep an eye on resource usage & parallel jobs because with the integration tests this could get reasonably large. 

With some discussion with @Molter73 we think we can reduce the integration tests down to a single job by combining VM creation in Ansible and running those over just two jobs rather than the current eight. I'll tackle this in a follow up PR if it becomes a problem